weaver/core/drivers/fabric-driver/server/satp.ts

```diff
@@ -0,0 +1,363 @@
+    networkName: string
+): Promise<any> {
+
+    // TODO: remove the hardcoded values 
+    let performLockRequest2 = {};
+    performLockRequest2['target-network'] = 'network1';
+    performLockRequest2['hashBase64'] = 'ivHErp1x4bJDKuRo6L5bApO/DdoyD/dG0mAZrzLZEIs=';
+    performLockRequest2['timeout-duration'] = parseInt('3600');
+    performLockRequest2['locker'] = 'alice';
+    performLockRequest2['recipient'] = 'bob';
+    performLockRequest2['lockerWalletPath'] = '../wallet-network1/alice.id';
+    performLockRequest2['recipientWalletPath'] = '../wallet-network1/bob.id';
+    performLockRequest2['param'] = 'bond01:a05';
+    performLockRequest2['channel'] = 'mychannel';
+    performLockRequest2['chaincode-id'] = 'satpsimpleasset';
+
+    // Locker and recipient
+    const locker = performLockRequest2['locker'];
+    const recipient = performLockRequest2['recipient'];
+    const lockerWalletPath = performLockRequest2['lockerWalletPath'];
+    const recipientWalletPath = performLockRequest2['recipientWalletPath'];
+    let hashFn = performLockRequest2['hash_fn'];
+    let hashBase64 = performLockRequest2['hashBase64'];
+    const targetNetwork = performLockRequest2['target-network'];
+    const channel = performLockRequest2['channel'];
+    const chaincodeId = performLockRequest2['chaincode-id'];
+
+    // Hash
+    let hash: HashFunctions.Hash
+    if (hashFn == 'SHA512') {
+        hash = new HashFunctions.SHA512()
+    } else {
+        hash = new HashFunctions.SHA256()
+    }
+
+    if (hashBase64) {
+        hash.setSerializedHashBase64(hashBase64)
+    }
+    else {
+        logger.info(`No hash provided, using random preimage`)
+    }
+
+    // Timeout
+    var timeout = 0, timeout2 = 0;
+    const currTime = Math.floor(Date.now() / 1000);
+    if (performLockRequest2['timeout-epoch']) {
+        let duration = performLockRequest2['timeout-epoch'] - currTime
+        timeout = performLockRequest2['timeout-epoch']
+        timeout2 = performLockRequest2['timeout-epoch'] + duration
+    }
+    else if (performLockRequest2['timeout-duration']) {
+        timeout = currTime + performLockRequest2['timeout-duration']
+        timeout2 = currTime + 2 * performLockRequest2['timeout-duration']
+    }
+
+    let gateway: Gateway = await getNetworkGateway(networkName);
+    const network: Network = await gateway.getNetwork(channel);
+    const contract = network.getContract(chaincodeId);
+
+    const params = performLockRequest2['param'].split(':')
+    const lockerCert = Buffer.from(lockerWalletPath).toString('base64')
+    const recipientCert = Buffer.from(recipientWalletPath).toString('base64')
+
+    var funcToCall, asset
+
+    if (performLockRequest2['fungible']) {
+        funcToCall = SatpAssetManager.createFungibleHTLC
+        asset = 'Fungible Asset'
+    } else {
+        funcToCall = SatpAssetManager.createHTLC
+        asset = 'Asset'
+    }
+
+    console.info(`Asset Lock: Lock ${asset}:\n`);
```

Comment:
Use logger instead of console log (applies to entire PR)

Reply:
@outsidethecode Please replace all `console.log` invocations with `logger` calls.

Reply:
Fixed.